fecgov/openFEC
# Upgrade node and npm packages to latest

## Summary 

Upgrade NODE package from v10 to LTS v14.13.0
Upgrade legacy NPM to LTS v7.23.0

- Resolves #4676

1.  update node and npm package versions in package.json file
2.  update node version  in circleci/config.xml

## Before upgrade:
```
(4149-env) macadmins-MacBook-Pro-4:openFEC pkasireddy$ node -v
v10.16.0
(4149-env) macadmins-MacBook-Pro-4:openFEC pkasireddy$ npm -v
6.13.7
```

## After upgrade:
```
(api3710) macadmins-mbp-5:openFEC pkasireddy$ node -v
v14.13.0
(api3710) macadmins-mbp-5:openFEC pkasireddy$ npm -v
7.23.0
```

### Required reviewers
At least two developers

## How to test
- checkout branch
- create new python virtual env: run `pyenv virtualenv 3.7.10 api3710` 
- install requirements: run `pip install -r requirements.txt -r requirements-dev.txt`
- check the local  node and npm version before upgrading:
    - run `node -v` :  terminal  display `10.16.0`
    - run `npm -v` : terminal  display `6.13.7`
- remove existing node_modules folder on openFEC repo: run `rm -rf node-modules`
- display existing nvm verisons available on local: run `nvm ls`
- if nvm v14.13.0 doesnt exist on local already run : `nvm install v14.13.0`
- default the nvm version to 14.13.0: run `nvm alias default v14.13.0`
- now verify the node version:  run `node -v `: terminal display `14.13.0`
- clear npm cache: run `npm cache clean -f`
- install/update to latest npm:  run `npm install -g npm@7.23.0`
- verify the npm version: run `npm -v` : terminal display `7.23.0`
- run `npm i`
- run `npm run build`
- run `pytest`, just in case
- start server and test api endpoints on swagger (http://127.0.0.1:5000/)
